pennylane/devices/tests/test_measurements.py

```diff
@@ -354,6 +354,9 @@ def test_sample_values(self, device, tol):
         n_wires = 1
         dev = device(n_wires)
 
+        if dev.shots is None:
+            pytest.skip("Device is in analytic mode, cannot test sampling.")
```

Comment:
Good to be aware of: the `sample` return type is not tested on analytic devices any more

Reply:
Hm that seems bad. Do we test analytic devices with `shots=None` and `shots=shots_num`?

Reply:
@Thenerdstation, this is exactly what we have to do from now on. I am not too sure though, @Josh mentioned that this increases the test time by quite a lot. But at least the plugin-test-matrix should have both options! 

@thisac I don't know...I thought in case of deprecation we test the new, recommended behaviour as well as the warning (since we don't guarantee the deprecated behaviour to work)?

Reply:
I'm not sure about best practices here either to be honest. I noticed the warnings test further down, and, since we're deprecating this (and will switch it to an error in the hopefully not-too-far-away future) I think it might be fine not testing it.

Maybe include something like "Sampling deprecated for analytic devices and might currently cause undefined behaviour" in the warning, although it might not be necessary. 🤔